crates/turborepo-lib/src/run/pipeline.rs

```diff
@@ -0,0 +1,100 @@
+use std::collections::{HashMap, HashSet};
+
+use serde::{Deserialize, Serialize};
+
+pub type Pipeline = HashMap<String, BookkeepingTaskDefinition>;
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct BookkeepingTaskDefinition {
```

Comment:
cc @mehulkar 

Is there a path to dropping the `Bookkeeping` in a rust world? Post-port? Or something we can do as part of the porting effort?

Reply:
It's main goal was to assist in merging fields. If Rust has a better way to distinguish between when a field is missing vs when it's the "empty value", we should be able to drop it. The [integration tests covering this behavior][1] are pretty good, so I'd just say "try it" 

[1]: https://github.com/vercel/turbo/tree/71cb29adf6b4b05a0a08ccb8fd1cf561a97f2670/turborepo-tests/integration/tests/composable_config

Reply:
Yep! We can do `Option<T>` for those